TrackerExtenstion.cs

```diff
@@ -38,14 +39,48 @@ public void Startup(StartupParams sp)
         /// </param>
         public void Ready(ReadyParams rp)
         {
+            // @todo Subscribe to an event after the Dynamo home screen is shown
+
+            // Display a MessageBox to the user
+            // @todo Use a view framework to improve the UI/UX
             string message = "By pressing OK you agreeing to Binoculars 🔍 data collection.";
             string title = "Terms of Use Agreement";
-
             MessageBox.Show(message, title, MessageBoxButton.OK, MessageBoxImage.Exclamation);
+
+            // @todo if user does not consent, don't store..
+            if (true) {
+
+                // Dynamo StartupParams
+                ExportData.dynamoVersion = rp.StartupParams.DynamoVersion.ToString();
+
+                // @todo Determine whether the script is run from DynamoPlayer..
+
+                // Store the local environment data
+                ExportData.user = Environment.MachineName;
+                ExportData.computerName = Environment.UserName;
+                
+                // @todo Define the revitVersion, leave blank or null if opened from any other environment (Sandbox, Civil3D etc)
+                ExportData.revitVersion = "2019.0.2";
+
+                // Request the IP and geolocation from ipinfo.io
+                WebClient client = new WebClient();
+                client.Headers.Set("Accept", "application/json");
+                var json = client.DownloadString("https://ipinfo.io");
```

Comment:
wouldn't put this in the `Ready` event, since it'd be better if it was deferred to be executed when needed - which is not on startup, but on event collection or export

Reply:
I'm currently measuring the response time of ipinfo.io as between 180 and 300ms on a decent connection which is quite significant, hence why I'd rather it was completed during Dynamo startup rather than on EvaluationCompleted. The presumption is that anyone loading Dynamo will at some point run a graph so deferring it makes little difference but slowing down their first run will be noticed and should be minimised.